apps/portals/e2e/explore.spec.ts

```diff
@@ -108,6 +113,70 @@ const getAvailableFacetsDiv = (page: Page) => {
     .locator('> div')
     .filter({ hasText: 'Available Facets' })
 }
+const toggleChartVisibilityAndVerifyCount = async (page: Page) => {
+  const chartsPanel = page.locator('.FacetNav')
+
+  const allChartCells = chartsPanel.locator('.FacetNavPanel')
+  const allChartsCount = await allChartCells.count()
+
+  const chartCells = chartsPanel.locator('.FacetNavPanel:visible')
+  const initialCount = await chartCells.count()
+  await test.step('confirm chat visibility and count', async () => {
+    let foundChartIndex = -1
+    for (let i = 0; i < allChartsCount; i++) {
+      const chart = allChartCells.nth(i)
+      if (await chart.isVisible()) {
+        foundChartIndex = i
+        break
+      }
+    }
+    const chartCell = allChartCells.nth(foundChartIndex)
+    const chartTitle = chartCell.locator('.FacetNavPanel__title')
```

Comment:
It's preferable to use roles/labels vs CSS classes to find UI elements on a page, especially in E2E tests. For some of these deeply-nested selectors, they should basically be treated as implementation details where they can be changed/removed at any time, which should minimally impact these tests.

What do you think of this approach instead?
1. Add a prop `role="toolbar"` to [the div](https://github.com/Sage-Bionetworks/synapse-web-monorepo/blob/8d098093961c76e042310e27ab3a930e7b1cb98c/packages/synapse-react-client/src/components/widgets/facet-nav/FacetNavPanel.tsx#L362) that contains these buttons (currently has the class `FacetNavPanel__title__tools`)
2. Replace this query with something like this (note: I am not sure if you can mix `getByRole` with `locator`, but hopefully something like this works)
```suggestion
    const chartToolbar = await chartCell.getByRole('toolbar')
```

Reply:
I want to acknowledge that some of the tests in this file are doing this, so it's understandable that you just followed by example